src/handlers/buyselltrade.ts

```diff
@@ -0,0 +1,138 @@
+import { Message, Client, TextChannel, MessageEmbed } from "discord.js";
+import { clearTimeout } from "timers";
+import config from '../config';
+
+class BstPost extends Message {
+    validationMessages: string[];
+    // folowing only set when validation fails 
+    warningMessage?: Message;
+    expiration?: NodeJS.Timeout;
+
+    constructor(msg: Message, warningMsg?: Message, expiration?: NodeJS.Timeout) {
+        super(msg.client, {
+            id: msg.id,
+            type: msg.type,
+            content: msg.content,
+            author: msg.author,
+            pinned: msg.pinned,
+            tts: msg.tts,
+            embeds: msg.embeds,
+            attachments: msg.attachments,
+            nonce: msg.nonce
+        }
+            , msg.channel);
+
+        this.validationMessages = [];
+        this.warningMessage = warningMsg;
+        this.expiration = expiration;
+    }
+    addMessage(errText: string) {
+        this.validationMessages.push(errText);
+    }
+    setMessageExpiration(timeInMs: number) {
+        setTimeout(this.deleteMessage.bind(this), timeInMs);
+    }
+    async deleteMessage() {
+        await this.delete();
+        this.deleted = true;
+        await this.warningMessage?.edit(this.warningMessage.content + "**Post was not edited in 30 minutes and was deleted**");
+    }
+    clearMessageExpiration() {
+        if (this.expiration)
+            clearTimeout(this.expiration);
+    }
+}
+
+export default async function handleBuySellTradeMessage(
+    msg: Message,
+    client: Client
+): Promise<void> {
+    if (msg.channel.id == config.FORTIES_BST) {
+        const currPost = new BstPost(msg);
+        validateBstMessage(currPost);
+    }
+}
+
+async function validateBstMessage(bstPost: BstPost) {
+    // add any additional validations here 
+    if (validateLocationBlockExistance(bstPost)) {
+        validateLocationBlockLength(bstPost);
+    }
+
+    if (bstPost.validationMessages.length) {
+        await handleFailedBstValidation(bstPost);
+    }
+    if (bstPost.warningMessage) {
+        // if this isn't a messages first time through the system, clear the delete timer
+        bstPost.clearMessageExpiration();
+    }
+}
+
+async function handleFailedBstValidation(bstPost: BstPost) {
+    const channel = bstPost.guild?.channels.cache.get(config.FORTIES_BST_DISCUSSION);
```

Comment:
Since we know the BST-discussion channel is a `TextChannel` we can avoid the weirdness here by just doing

`bstPost.guild?.channels.cache.get(config.FORTIES_BST_DISCUSSION) as TextChannel`